### PR TITLE
Clean up `cd.rs`

### PR DESCRIPTION
**Pull request by rgwood**
Some general cleanup of `cd.rs`; the permission checking code was a little hard to follow. Reworded comments and variable names, reorganized+renamed the module used for Unix file permissions.

### CODE REVIEW

File: crates/nu-command/src/filesystem/cd.rs [hunk 1]
```diff
@@ -10,22 +10,16 @@ use nu_protocol::{
...
-    }
+    pub const USER_EXECUTE: Mode = libc::S_IXUSR as Mode;
+    pub const GROUP_EXECUTE: Mode = libc::S_IXGRP as Mode;
+    pub const OTHER_EXECUTE: Mode = libc::S_IXOTH as Mode;
 }
 
-// use to return the message of the result of change director
-// TODO: windows, maybe should use file_attributes function in https://doc.rust-lang.org/std/os/windows/fs/trait.MetadataExt.html
-// TODO: the meaning of the result of the function can be found in https://learn.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants
-// TODO: if have realize the logic on windows, remove the cfg
```
Comment (sholderbach) [inline crates/nu-command/src/filesystem/cd.rs:28]: Is there a good place for those windows related resources?

Comment (rgwood) [reply crates/nu-command/src/filesystem/cd.rs:28]: I rewrote those comments and moved them to a place where I think they make more sense. The links are still in the file, just further down. Does that answer your question?

Comment (sholderbach) [reply crates/nu-command/src/filesystem/cd.rs:28]: Ah didn't spot it correctly through the rewording. All good!
